addons/account/models/account_move.py

```diff
@@ -1413,31 +1413,56 @@ def _compute_partner_credit_warning(self):
                            move.company_id.account_use_credit_limit
             if show_warning:
                 move.partner_credit_warning = self._build_credit_warning_message(
-                    move, move.partner_credit, move.tax_totals['amount_total'] > 0.0)
+                    move,
+                    current_amount=move.tax_totals['amount_total'],
+                    exclude_current=True,
+                )
 
     @api.depends('partner_id')
     def _compute_partner_credit(self):
         for move in self:
             move.partner_credit = move.partner_id.commercial_partner_id.credit
 
-    def _build_credit_warning_message(self, record, updated_credit, include):
-        """ Build the warning message that will be displayed in a yellow banner on top of the current record
-            if the partner exceeds a credit limit (set on the company or the partner itself).
-            :param record:                  The record where the warning will appear (Invoice, Sales Order...).
-            :param updated_credit (float):  The partner's updated credit limit including the current record.
-            :param include (bool):          Whether the current record's amount is included in the warning message.
-            :return (str):                  The warning message to be showed.
+    def _build_credit_warning_message(self, record, current_amount=0.0, exclude_current=False):
+        """Build the warning message that will be displayed in a yellow banner on top of the current record if the
+        partner exceeds a credit limit (set on the company or the partner itself).
+
+        :param record:                    The record where the warning will appear (Invoice, Sales Order...)
+        :param current_amount (float):    The partner's outstanding credit amount from the current document
+        :param exclude_current (bool):    Whether to exclude `current_amount` from the credit to invoice
+        :return (str):                    The warning message to be showed
```

Comment:
Only 2 lines have changed in this block, please keep the diff to a minimum :nerd_face: